VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/NotificationsSettingsScreen/NotificationsSettingsScreen.tsx

```diff
@@ -22,90 +34,129 @@ import {
     Alert.alert(t('leavingApp.title'), t('leavingApp.body.settings'), [
       {
         text: t('leavingApp.cancel'),
         style: 'cancel',
       },
-      { text: t('leavingApp.ok'), onPress: () => Linking.openSettings(), style: 'default' },
+      { text: t('leavingApp.ok'), onPress: openSettings, style: 'default' },
     ])
   }
-  const dispatch = useAppDispatch()
 
-  useOnResumeForeground(() => {
-    if (deviceToken) {
-      dispatch(loadPushPreferences(ScreenIDTypesConstants.NOTIFICATIONS_SETTINGS_SCREEN))
+  const fetchPreferences = async () => {
+    const endpoint_sid = await AsyncStorage.getItem(DEVICE_ENDPOINT_SID)
+    const deviceToken = await AsyncStorage.getItem(DEVICE_TOKEN_KEY)
+
+    if (endpoint_sid && deviceToken) {
+      refetchPushPreferences()
     } else {
       Notifications.events().registerRemoteNotificationsRegistered((event) => {
-        dispatch(registerDevice(event.deviceToken, true))
+        const registerParams = {
+          deviceToken: event.deviceToken,
+        }
+        const mutateOptions: MutateOptions<PushRegistrationResponse | undefined, Error, RegisterDeviceParams, unknown> =
+          {
+            onSettled: () => {
+              refetchPushPreferences()
+            },
+          }
+        registerDevice(registerParams, mutateOptions)
       })
       Notifications.events().registerRemoteNotificationsRegistrationFailed(() => {
-        dispatch(registerDevice())
+        const registerParams = {
+          deviceToken: undefined,
+        }
+        const mutateOptions: MutateOptions<PushRegistrationResponse | undefined, Error, RegisterDeviceParams, unknown> =
+          {
+            onSettled: () => {
+              refetchPushPreferences()
+            },
+          }
+        registerDevice(registerParams, mutateOptions)
       })
       Notifications.registerRemoteNotifications()
     }
-  })
+  }
 
-  useEffect(() => {
-    if (screenContentAllowed('WG_NotificationsSettings')) {
-      dispatch(loadPushPreferences(ScreenIDTypesConstants.NOTIFICATIONS_SETTINGS_SCREEN))
-    }
-  }, [dispatch])
+  useOnResumeForeground(fetchPreferences)
 
   const preferenceList = (): ReactNode => {
-    const prefsItems = preferences.map((pref): SimpleListItemObj => {
-      return {
-        a11yHintText: t('notifications.settings.switch.a11yHint', { notificationChannelName: pref.preferenceName }),
-        text: pref.preferenceName,
-        decorator: ButtonDecoratorType.Switch,
-        decoratorProps: {
-          on: pref.value,
-        },
-        onPress: () => {
-          logAnalyticsEvent(Events.vama_toggle(pref.preferenceName, !pref.value, t('notifications.title')))
-          dispatch(setPushPref(pref))
-        },
-      }
-    })
-    return (
-      <Box mt={condensedMarginBetween}>
-        <SimpleList items={prefsItems} />
-      </Box>
-    )
+    if (pushPreferences) {
```

Comment:
Minor suggestion, you could have an `if` statement that returns early if `pushPreferences` doesn't exist instead of wrapping all the rendering logic in an if statement, e.g.

```tsx
if (!pushPreferences) return
```

Reply:
updated